components/tools/OmeroPy/test/integration/test_model51.py

```diff
@@ -100,3 +100,55 @@ def testAsMapMethod(self):
         )
         m = g.getConfigAsMap()
         assert m["foo"] == "bar"
+
+    def assertMapAnnotations(self, anns, length=2):
+        m = None
+        assert length == len(anns)
+        for idx in range(length):
+            a = anns[idx]
+            if isinstance(a, omero.model.MapAnnotationI):
+                m = anns[idx]
+        assert m
+        assert "foo" == m.getMapValue()[0].name
+        assert "bar" == m.getMapValue()[0].value
+
+    def testMapEagerFetch(self):
+        m = omero.model.MapAnnotationI()
+        m.setMapValue(
+            [NV("foo", "bar")]
+        )
+        m = self.update.saveAndReturnObject(m)
+        anns = self.query.findAllByQuery(
+            "select m from MapAnnotation m ",
+            None)
```

Comment:
This should be a new user in a new group, but I'm happy to add filtering. I wasn't quite sure of the status of @sbesson's re-working to know what the new "Right Way" was (to be).